src/core/image.rs

```diff
@@ -26,70 +37,32 @@ where
     /// users will either go with the default arguments or just override one or two. When defining
     /// the arguments of your image, consider that the whole purpose is to facilitate integration
     /// testing. Only expose those that actually make sense for this case.
-    type Args;
+    fn args(&self) -> Box<dyn Iterator<Item = String>> {
+        Box::new(vec![].into_iter())
+    }
 
-    /// A type representing the environment variables for an Image.
-    ///
-    /// There are a couple of things regarding the arguments of images:
+    /// There are a couple of things regarding the environment variables of images:
     ///
     /// 1. Similar to the Default implementation of an Image, the Default instance
     /// of its environment variables should be meaningful!
     /// 2. Implementations should be conservative about which environment variables they expose. Many times,
     /// users will either go with the default ones or just override one or two. When defining
     /// the environment variables of your image, consider that the whole purpose is to facilitate integration
     /// testing. Only expose those that actually make sense for this case.
-    type EnvVars;
+    fn env_vars(&self) -> Box<dyn Iterator<Item = (&String, &String)> + '_> {
+        Box::new(std::iter::empty())
+    }
```

Comment:
Regardless of where this method and `volumes()` ends up, shall we also then avoid iterators as their return types? Simply just `BTreeMap` or `HashMap`?

Reply:
Not too sure about exposing maps directly. The most UX friendly might be to create our own type, plus a macro or other convenient APIs to declare them effortlessly. Initially, I settled on iterators to be able to make more changes without them being breaking.

Reply:
If it makes things easier, we can roll with exposing the maps directly though!